tools/vfsstat.py

```diff
@@ -2,39 +2,47 @@
 # @lint-avoid-python-3-compatibility-imports
 #
 # vfsstat   Count some VFS calls.
-#           For Linux, uses BCC, eBPF. See .c file.
+#           For Linux, uses BCC, eBPF. Embedded C.
 #
 # Written as a basic example of counting multiple events as a stat tool.
 #
-# USAGE: vfsstat [interval [count]]
+# USAGE: vfsstat [-h] [-p PID] [interval] [count]
 #
 # Copyright (c) 2015 Brendan Gregg.
 # Licensed under the Apache License, Version 2.0 (the "License")
 #
 # 14-Aug-2015   Brendan Gregg   Created this.
+# 12-Oct-2022   Rocky Xing      Added PID filter support.
 
 from __future__ import print_function
 from bcc import BPF
 from ctypes import c_int
 from time import sleep, strftime
 from sys import argv
-
-def usage():
-    print("USAGE: %s [interval [count]]" % argv[0])
-    exit()
+import argparse
 
 # arguments
-interval = 1
-count = -1
-if len(argv) > 1:
-    try:
-        interval = int(argv[1])
-        if interval == 0:
-            raise
-        if len(argv) > 2:
-            count = int(argv[2])
-    except:  # also catches -h, --help
-        usage()
+examples = """examples:
+    ./vfsstat             # count some VFS calls
+    ./vfsstat -p 185      # trace PID 185 only
+    ./vfsstat 1 10        # print 1 second summaries, 10 times
+"""
+parser = argparse.ArgumentParser(
+    description="Count some VFS calls.",
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    epilog=examples)
+parser.add_argument("-p", "--pid",
+    help="trace this PID only")
+parser.add_argument("interval", nargs="?", default=99999999,
```

Comment:
Let's keep the default interval to 1, so that it won't change the behavior of the tool. Other than that, looks good to me.

Reply:
ok, it's my fault.